.test-infra/jenkins/job_Publish_SDK_Image_Snapshots.groovy

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import CommonJobProperties as commonJobProperties
+
+// This job publishes regular snapshots of the SDK harness containers for
+// testing purposes. It builds and pushes the SDK container to the
+// specified GCR repo, tagged at the current Git commit.
+job('beam_Publish_Beam_SDK_Snapshots') {
+  description('Builds SDK harness images snapshots regularly for testing purposes.')
+
+  // Set common parameters.
+  commonJobProperties.setTopLevelMainJobProperties(delegate)
+
+  // // Runs once per hour.
+  commonJobProperties.setAutoJob(delegate, '0 * * * *')
+
+  // Use jenkins env var interpolation - leave in single quotes
+  def imageRepo = 'gcr.io/apache-beam-testing/beam-sdk'
+  def imageTag = '${GIT_COMMIT}'
```

Comment:
If there is a better place to pull this in, please let me know!

Reply:
I also wonder if we should error if not running against master - it would make sense to at least tag it differently or place non-master in a different repository folder so a PR-created snapshot can't replace a valid master snapshot as a latest tag.

Reply:
if this job is triggered on schedule, will it actually be run against non-master? or we talk about somebody trying to run the job manually against non-master?

Reply:
the second - basically to guard against test snapshots getting put into the same repo. Our plan is to tie the import to a given snapshot commit, so whoever is doing the import will want to choose a commit at which a snapshot exists.

Reply:
I think there is no concern with on-demand runs since we tag on git commit. If we will need to build these container images for handling non-master imports, we can include branch name in the imageRepo or use the release/rc tags instead of commit tags.